syncstorage/src/db/mod.rs

```diff
@@ -69,13 +77,24 @@ pub fn spawn_pool_periodic_reporter<T: GetPoolState + Send + 'static>(
     Ok(())
 }
 
-pub async fn run_on_blocking_threadpool<F, T>(f: F) -> Result<T, DbError>
+/// Runs a function as a task on Actix's blocking threadpool.
+///
+/// WARNING: Calling `web::block` anywhere else will result in inaccurate threadpool metrics being
+/// reported. If you want to spawn a task on Actix's blocking threadpool, you **must** use this
+/// function.
+pub async fn run_on_blocking_threadpool<F, T>(metrics: Metrics, f: F) -> Result<T, DbError>
```

Comment:
I don't love that we need to pass the `Metrics` into the function, but I couldn't think of a clean way to avoid it